packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -32,92 +36,153 @@ export class FluidSerializer implements IFluidSerializer {
         bind: IFluidHandle,
     ) {
         // If the given 'input' cannot contain handles, return it immediately.  Otherwise,
-        // return the result of 'recursivelyReplaceHandles()'.
-        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions,@typescript-eslint/no-unsafe-return
+        // return the result of 'recursivelyEncode()'.
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
         return !!input && typeof input === "object"
-            ? this.recursivelyReplaceHandles(input, bind)
+            ? this.recursivelyEncode(input, bind)
             : input;
     }
 
+    // TODO: The below proof-of-concept implementation of 'decode()' will be enabled with tests
+    //       and benchmarks once the refactoring to use EJS codegen lands.
+    /*
+    public decode(
+        input: any,
+    ) {
+        // If the given 'input' cannot contain handles, return it immediately.  Otherwise,
+        // return the result of 'recursivelyDecode()'.
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        return !!input && typeof input === "object"
+            ? this.recursivelyDecode(input)
+            : input;
+    }
+    */
+
     public stringify(input: any, bind: IFluidHandle) {
         return JSON.stringify(input, (key, value) => {
-            // If the current 'value' is not a handle, return it unmodified.  Otherwise,
-            // return the result of 'serializeHandle'.
             // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-            const handle = !!value && value.IFluidHandle;
-            // TODO - understand why handle === false in some of our tests
```

Comment:
FYI - I suspect the original 'TODO' was the a mistake along the lines of 'return !!value && value.IFluidHandle', which returns 'false' due to the coercion.